docs/recipes/install/rocky9/x86_64/openchami/slurm/recipe.sh

```diff
@@ -0,0 +1,663 @@
+# Install Development Tools 
+# --------------------------
+dnf -y install ohpc-autotools
+dnf -y install EasyBuild-ohpc
+dnf -y install hwloc-ohpc
+dnf -y install spack-ohpc
+dnf -y install valgrind-ohpc
+
+# ------------------
+# Install Compilers 
+# ------------------
+dnf -y install gnu14-compilers-ohpc
+
+# -------------------
+# Install MPI Stacks 
+# -------------------
+if [[ ${enable_mpi_defaults} -eq 1 ]];then
+     dnf -y install openmpi5-pmix-gnu14-ohpc mpich-ofi-gnu14-ohpc
+fi
+
+if [[ ${enable_ib} -eq 1 ]];then
+     dnf -y install mvapich2-gnu14-ohpc
+fi
+if [[ ${enable_opa} -eq 1 ]];then
+     dnf -y install mvapich2-psm2-gnu14-ohpc
+fi
+
+# --------------------------
+# Install Performance Tools 
+# --------------------------
+dnf -y install ohpc-gnu14-perf-tools
+dnf -y install lmod-defaults-gnu14-openmpi5-ohpc
+
+# --------------------------------------
+# Install 3rd Party Libraries and Tools 
+# --------------------------------------
+dnf -y install ohpc-gnu14-serial-libs
+dnf -y install ohpc-gnu14-io-libs
+dnf -y install ohpc-gnu14-python-libs
+dnf -y install ohpc-gnu14-runtimes
+if [[ ${enable_mpi_defaults} -eq 1 ]];then
+     dnf -y install ohpc-gnu14-mpich-parallel-libs
+     dnf -y install ohpc-gnu14-openmpi5-parallel-libs
+fi
+if [[ ${enable_ib} -eq 1 ]];then
+     dnf -y install ohpc-gnu14-mvapich2-parallel-libs
+fi
+if [[ ${enable_opa} -eq 1 ]];then
+     dnf -y install ohpc-gnu14-mvapich2-parallel-libs
+fi
+
+# ---------------------------
+# Install Intel oneAPI tools 
+# ---------------------------
+if [[ ${enable_intel_packages} -eq 1 ]];then
+     dnf -y install intel-oneapi-toolkit-release-ohpc
+     rpm --import https://yum.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB
+     dnf -y install intel-compilers-devel-ohpc
+     dnf -y install intel-mpi-devel-ohpc
+     if [[ ${enable_opa} -eq 1 ]];then
+          dnf -y install mvapich2-psm2-intel-ohpc
+     fi
+     dnf -y install openmpi5-pmix-intel-ohpc
+     dnf -y install ohpc-intel-serial-libs
+     dnf -y install ohpc-intel-geopm
+     dnf -y install ohpc-intel-io-libs
+     dnf -y install ohpc-intel-perf-tools
+     dnf -y install ohpc-intel-python3-libs
+     dnf -y install ohpc-intel-mpich-parallel-libs
+     dnf -y install ohpc-intel-mvapich2-parallel-libs
+     dnf -y install ohpc-intel-openmpi5-parallel-libs
+     dnf -y install ohpc-intel-impi-parallel-libs
+fi
+
+# -------------------------------------------------------------
+# Allow for optional sleep to wait for provisioning to complete
+# -------------------------------------------------------------
+sleep ${provision_wait}
+
+# -------------------------
+# Resource Manager Startup 
+# -------------------------
+systemctl enable munge
+systemctl enable slurmctld
+systemctl start munge
+systemctl start slurmctld
+pdsh -w ${compute_prefix}[1-${num_computes}] systemctl start munge
+pdsh -w ${compute_prefix}[1-${num_computes}] systemctl start slurmd
+
+# ----------
```

Comment:
The cloudinit part should happen before the line `sleep ${provision_wait}`. This is where we wait for the computes to boot and after that sleep the compute nodes should work. There is also a tex file called `reset-computes-*tex`. The recipe should include the step to reboot all compute nodes so that they are up. This is still missing.